test/role.test.js

```diff
@@ -87,9 +84,24 @@ describe('role model', function() {
     });
   });
 
+  it('should not allow duplicate role name', function(done) {
+    Role.create({ name: 'userRole' }, function(err, role) {
+      if (err) return done(err);
+
+      Role.create({ name: 'userRole' }, function(err, role) {
+        expect(err).to.exist;
+        expect(err).to.have.property('name', 'ValidationError');
+        expect(err).to.have.property('message');
+        expect(err.message).to.have.string('`name` already exists');
```

Comment:
Rather than checking exact error message, which may change in the future and thus break this test, it's better to check the error code, which should remain the same. The code can be found in the array in `error.details.codes.name`.

You can print the error object to see what I mean, I think it should look like this:

```js
{
  name: 'ValidationError',
  message: '`name` already exists',
  details: {
    codes: {
      name: [
        'uniquenes'
      ]
    }
  },
  messages: {
    name: [
      'already exists'
    ]
  }
}
```